### ISSUE TITLE
"Hello, we've noticed an issue with your iOS app. Please delete this app and re-install it from the App Store."

### ISSUE THREAD

**Issue by petrchutny** [stephenou/fruitionsite#242]
My website using Fruition now cannot be displayed, showing this error in a modal box. The same goes for the official Fruition website.

"Hello, we've noticed an issue with your iOS app. Please delete this app and re-install it from the App Store."

![Error](https://github.com/stephenou/fruitionsite/assets/1197393/09615636-8056-4301-a340-978b91a95ce4)

**Comment by thisismikovic** [comment]
I was randomly trying out stuff - when I completely remove MetaRewriter class (also from the bottom of the js file), this error will disappear and site will load normally. Someone smarter should tell us why, as I accidentally discovered it.

P.S. removing MetaRewriter also solved #241  for me

**Comment by DungLai** [comment]
> I was randomly trying out stuff - when I completely remove MetaRewriter class (also from the bottom of the js file), this error will disappear and site will load normally. Someone smarter should tell us why, as I accidentally discovered it.
> 
> P.S. removing MetaRewriter also solved #241 for me

I tried this but it doesn't work for me

**Comment by DungLai** [comment]
> I was randomly trying out stuff - when I completely remove MetaRewriter class (also from the bottom of the js file), this error will disappear and site will load normally. Someone smarter should tell us why, as I accidentally discovered it.
> 
> P.S. removing MetaRewriter also solved #241 for me

This works for me, thank you so much! 

After you remove the MetaRewriter and redeploy, wait 3-5 minutes and try again. It won't work straight away for me.

**Comment by raznostoronniy** [comment]
It works @thisismikovic 

Removing all MetaRewriter class with 2 strings from the bottom:

      .on('title', new MetaRewriter())
      .on('meta', new MetaRewriter())

**Comment by thisismikovic** [comment]
@raznostoronniy yeah, but I think the guys found a better solution. Try this: https://github.com/stephenou/fruitionsite/pull/243
Thanks to @cn-kali-team

**Comment by vipul354750** [comment]
> @raznostoronniy yeah, but I think the guys found a better solution. Try this: #243 Thanks to @cn-kali-team

hey i tried this but wont work. the thread is quite confusing for me as he mentions the rust version somewhere. i tried modifying the code from the two commits there but still as a non coder and unfamiliarity with github ui, its quite challenging for me. can you pl specify which lines to remove and what to add!

**Comment by vipul354750** [comment]
> @raznostoronniy yeah, but I think the guys found a better solution. Try this: #243 Thanks to @cn-kali-team

hey i tried removing that metaRewriter class and those two lines in the end and waited quite some till, but no success. 

also i tried https://github.com/stephenou/fruitionsite/pull/243/files changing to this code still doesnt wotk. is anything more needed to be done in order to load databases in this?

**Comment by bflack** [comment]
Hey, I'm not quite following what I'm supposed to do with #243, specifically the cname bit. I put these values into my DNS settings, but it doesn't feel quite right?
<img width="700" alt="Screenshot 2023-09-04 at 12 30 12 PM" src="https://github.com/stephenou/fruitionsite/assets/4500592/111d7522-9052-4c65-9a53-5f5f0cd96788">

**Comment by jayviddy** [comment]
remove metawriter class with those 2 strings worked for me. haven't noticed any side effects yet... 🤞 

don't really get what to do with #243 either.

**Comment by angadsri** [comment]
Hopefully final edit / summary:

- commented out the MetaRewriter stuff: it fixes this #242 iOS issue for me reliably
- did not make my `PAGE_TITLE` blank: doing that seems to interfere with the above fix — instead made all my databases inline to fix #241 

***

Edit 2: Turns out given more time the iOS issue recurred, oof. Now it looks like the MetaRewriter stuff being commented out is perhaps incompatible with the 'PAGE_TITLE' being made blank — I get Error 1101 now on Cloudflare and the site doesn't even load. 

Now that I have all my databases inline rather than full-page, I will check if #241 still happens while making the `PAGE_TITLE` no longer blank. 

***

Edit: realised I had this #241 error too so I reverted this fix for now, and tried @sanbotto's [suggested solution](https://github.com/stephenou/fruitionsite/issues/241#issuecomment-1699964616) to make the `PAGE_TITLE` blank. That seems to bring back my gallery views and also fix this iOS browser problem.

***

Original comment: Worked for me too! But just to be able to revert this in the future, I commented out the MetaRewriter bits :)

![image](https://github.com/stephenou/fruitionsite/assets/9767464/9d25d432-da26-45d7-9608-817266baa9d1)
![image](https://github.com/stephenou/fruitionsite/assets/9767464/ca190a03-91e0-4972-b068-81f45441b4bb)

**Comment by nguyenquocviet99** [comment]
> Edit: realised I had this #241 error too so I reverted this fix for now, and tried @sanbotto's [suggested solution](https://github.com/stephenou/fruitionsite/issues/241#issuecomment-1699964616) to make the `PAGE_TITLE` blank. That seems to bring back my gallery views and also fix this iOS browser problem. As for the top-left breadcrumbs being missing, #247 suggests that is a Notion-wide change for public pages, so will play around with manually adding the Breadcrumb block to all my pages (sigh).
> 
> Original comment: Worked for me too! But just to be able to revert this in the future, I commented out the MetaRewriter bits :)
> 
> ![image](https://user-images.githubusercontent.com/9767464/266352458-9d25d432-da26-45d7-9608-817266baa9d1.png) ![image](https://user-images.githubusercontent.com/9767464/266352524-ca190a03-91e0-4972-b068-81f45441b4bb.png)

![z4673046125215_fc10be9c3945d20bfa211a448f6990b1](https://github.com/stephenou/fruitionsite/assets/129487351/443c586b-f2bb-43a3-a820-99e5d4238812)
itsn't working for me

**Comment by dannykorpan** [comment]
Same problem here. Commenting the MetaRewriter class worked for me.

**Comment by disachantel** [comment]
is anyone still having this issue? removing the page title and MetaRewriter instances aren't helping. getting the iOS app prompt and databases still aren't showing.

**Comment by kiransyd** [comment]
I had to comment out these 3 lines for it to fully work:

```
      //.on('title', new MetaRewriter())
      //.on('meta', new MetaRewriter())
      //.on('head', new HeadRewriter())
```

**Comment by mdangear** [comment]
Same for me, removing just title and meta was not enough

> I had to comment out these 3 lines for it to fully work:
> 
> ```
>       //.on('title', new MetaRewriter())
>       //.on('meta', new MetaRewriter())
>       //.on('head', new HeadRewriter())
> ```

**Comment by maverck-** [comment]
> MetaRewriter

Thanks. This worked for me too. I only commented the class and it's related event handlers :)

**Comment by maverck-** [comment]
> I was randomly trying out stuff - when I completely remove MetaRewriter class (also from the bottom of the js file), this error will disappear and site will load normally. Someone smarter should tell us why, as I accidentally discovered it.
> 
> P.S. removing MetaRewriter also solved #241 for me

Thanks. This worked for me too. I only commented the class and it's related event handlers :)
